### PR TITLE
Compatibility with odoc.2.4.1

### PR DESCRIPTION
**Pull request by gpetiot**
odoc.2.3.0 should have been released already, should be there soon.

@sabine can this be moved to the staging branch once odoc.2.3.0 is released? (the CI pipeline needs to pin this version)

### CODE REVIEW

**Comment by gpetiot** [comment]
@sabine odoc.2.3.0 has now been released on opam. I cannot restart the CI but everything works locally with the new release.

**Comment by sabine** [comment]
CI restarted. There needs to be a corresponding PR on ocaml-docs-ci to pin odoc indeed. Ideally referencing back to this one, for clarity.

**Comment by gpetiot** [comment]
I opened a PR on ocaml-docs-ci: https://github.com/ocurrent/ocaml-docs-ci/pull/154

**Comment by tmcgilchrist** [comment]
What co-ordination is required for the deployment of this change with docs-ci and ocaml.org? 
Is ocaml.org backwards compatible with odoc.2.2.0 and 2.3.0 formats?

**Comment by tmcgilchrist** [comment]
@gpetiot Could you update both PRs to use 2.3.1 that includes some bug fixes?

**Comment by gpetiot** [comment]
I updated this PR and https://github.com/ocurrent/ocaml-docs-ci/pull/163 to use odoc.2.3.1

**Comment by gpetiot** [comment]
@sabine @EmileTrotignon I updated the PR for odoc.2.4.0 (the only change is the integration of https://github.com/ocaml/odoc/pull/949)

File: src/voodoo-gen/search_index.ml [hunk 1]
```diff
@@ -96,6 +113,17 @@ module Generate = struct
...
+          name :: full_name_aux (parent :> Identifier.t)
+      | `AssetFile (parent, name) ->
+          name :: full_name_aux (parent :> Identifier.t)
+      | `SourceLocationMod parent -> full_name_aux (parent :> Identifier.t)
+      | `SourceLocation (parent, name) ->
+          DefName.to_string name :: full_name_aux (parent :> Identifier.t)
+      | `SourceLocationInternal (parent, name) ->
+          LocalName.to_string name :: full_name_aux (parent :> Identifier.t)
+      | `SourcePage (parent, name) ->
+          name :: full_name_aux (parent :> Identifier.t)
```
Comment (EmileTrotignon) [inline src/voodoo-gen/search_index.ml:126]: This is fine, although a fullname function is now shipped by odoc, so it could be deleted.

**Comment by gpetiot** [comment]
Let's wait for `odoc.2.4.1` to be available on opam, as it's apparently necessary to have `eio` compile.

**Comment by gpetiot** [comment]
@sabine the CI is passing (except the experimental ones) on this PR, can we push it to the staging branch?

**Comment by sabine** [comment]
It's now on staging and will be picked up by staging.docs.ci.ocamllabs.io!

**Comment by sabine** [comment]
Seems to be okay on staging.docs.ci.ocamllabs.io, so merging this.

It will take a few days for [docs.ci.ocaml.org](http://docs.ci.ocaml.org/) to build a sufficient number of packages, so that we can be sure the deployment worked.

**Comment by art-w** [comment]
Did this accidentally change some urls? It looks like documentation is generated on the server as usual, but I get a 404 on every package doc on ocaml.org :(

(... edit: to clarify, I have no idea if this PR is related or if something else happened!)

**Comment by sabine** [comment]
This seems related but unintended because the documentation pipeline should be building docs in a new, non-live epoch. I passed this on to Mark.

**Pull request merged**